src/rtc_pl031.rs

```diff
@@ -147,8 +200,11 @@ impl RTC {
 
         match offset {
             RTCMR => {
-                // Set the match register, though this is not currently used.
+                // Set the match register.
+                // Since this is not currently used, we call invalid_write()
+                // from the rtc_events struct.
                 // TODO: Implement the match register functionality.
+                self.rtc_events.invalid_write();
```

Comment:
Hmm, I would differentiate this one from the invalid write because it isn't actually invalid, but rather unimplemented. How about we have a different metric for it (or no metric since for now we are not implementing the register on purpose)? Separating invalid writes from unimplemented registers could help us identify faulty drivers. Same should apply to read.

Reply:
Sure.  Firecracker uses `missed_read/write_count` for these and a mix of incrementing`error_count` metrics for some errors and returning an `Err` for others, which we don’t do here.

Do you have a preference between a read/write to the match register being ignored (from a metric standpoint) or incrementing a `missed_read/write` instead of an `invalid_read/write`?

Reply:
I just looked at the Firecracker code as well. We can have a missed_read/write metric separated from the invalid_read/write. This should give enough flexibility.

Reply:
Sounds good.  Presumably, a vmm's implementation of the `RTCEvents` trait can just leave the `missed_read/write` functions unimplemented (without cost?) if they only want metrics for actual `invalid_read/write`s...

Reply:
Yap, exactly! Having a separate metric for that is the best of both worlds because you can "emulate" both actions:
- a metric for when this happens
- do nothing when this happens because it's an expected behavior

Reply:
Updated.

The coverage test is upset that the default `NoEvents` implementation with empty metric methods doesn't get tested, hence the drop in coverage from 99.0% -> 97.5%.  Do you have a recommendation for fixing this?

I've attached a screen shot from kcov.  It thinks I cover only one of the four empty methods...

<img width="339" alt="Screenshot 2021-02-22 at 10 07 38" src="https://user-images.githubusercontent.com/43888469/108693854-41398e80-74f6-11eb-82c2-a1a23482f3e3.png">

Reply:
It looks like the only tests we could write are "coverage test" (i.e. test that are only useful for increasing the coverage). I am not personally in favor of such tests because of the burden of maintaining them. In this particular case, dropping the coverage seems the right approach to me.